### PR TITLE
Remove "from_email" default values

### LINKED ISSUE #1905: from_email should not has default values
In the config reference we have:

```
    from_email:
        address:        webmaster@example.com
        sender_name:    Admin
```

I think the default values should not be set (at least address), because nobody want to send mail from 'webmaster@example.com' and forgetting to set it may lead to unexpected errors (like email not sending because of the fake from address).

### PR DESCRIPTION
**Pull request by geoffrey-brier**
This PR fixes #1905

### CODE REVIEW

**Comment by XWB** [comment]
This will force a lot of users to update the configuration file. Not sure if that's an issue.

**Comment by geoffrey-brier** [comment]
I do agree ... maybe I could only write a warning note in the documentation about the default email address and sender name.

What do you think @XWB ?

**Comment by XWB** [comment]
We could make the email configuration optional and do not send an email if `from_email` is empty. However, that would change the behavior of the bundle.

**Comment by geoffrey-brier** [comment]
Not sure if it's the right solution. IMO, it would be better to force users to update their configuration instead of having them struggling to realize why their emails are not being sent anymore. Plus that would deeply change the behavior of the bundle ...

Any other ideas ?

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -54,10 +54,10 @@ public function getConfigTreeBuilder()
                 ->booleanNode('use_flash_notifications')->defaultTrue()->end()
                 ->booleanNode('use_username_form_type')->defaultTrue()->end()
                 ->arrayNode('from_email')
-                    ->addDefaultsIfNotSet()
+                    ->isRequired()
```
Comment (stof) [inline DependencyInjection/Configuration.php]: should not be required, as it is possible to use FOSUserBundle without sending emails (depending on the features you use)

Comment (geoffrey-brier) [inline DependencyInjection/Configuration.php]: Indeed, seems like I spoke too fast ... so this means that the top `from_email` node is just a helper node used to propogate its values to the `resetting` and `confirmation` nodes and it's not used by any other part of the bundle ?

Comment (XWB) [inline DependencyInjection/Configuration.php]: Looks like it. Maybe we can just remove the default webmaster values?

Comment (geoffrey-brier) [inline DependencyInjection/Configuration.php]: Yup, definitely doing that :+1:

**Comment by XWB** [comment]
@geoffrey-brier Tests still fail.

**Comment by geoffrey-brier** [comment]
@XWB Sorry I haven't been really active recently. I've reworked what I did so far, should be all right now ;)

File: DependencyInjection/FOSUserExtension.php [hunk 2]
```diff
@@ -106,20 +106,22 @@ public function load(array $configs, ContainerBuilder $container)
             ),
         ));
 
+        $fromEmail = isset($config['from_email']) ? $config['from_email'] : array();
```
Comment (XWB) [inline DependencyInjection/FOSUserExtension.php]: Logic like this should be handled by Symfony configuration.

**Comment by XWB** [comment]
@geoffrey-brier I think you are right, people should update their configuration. Have a look at https://github.com/FriendsOfSymfony/FOSUserBundle/pull/2417

**Comment by XWB** [comment]
https://github.com/FriendsOfSymfony/FOSUserBundle/pull/2417

**Pull request closed without merging**